snap/snapcraft.yaml

```diff
@@ -0,0 +1,28 @@
+name: urlscan
+version: git
+summary: Extract and browse the URLs contained in an email (urlview replacement)
+description: |
+  urlscan searches for URLs in email messages, then displays a list of
+  them in the current terminal.  It is primarily meant as a replacement
+  for urlview, which it improves upon in the following ways:
+    * urlscan understands email encodings such as quoted-printable;
+      urlview does not.
+    * urlscan extracts and displays the context surrounding each URL.
+
+grade: stable
+confinement: classic
+
+apps:
+  urlscan:
+    command: bin/urlscan-wrapper
+
+parts:
+  urlscan:
+    source: .
+    plugin: python
+    python-version: python3
+  urlscan-wrapper:
+    source: snap/
+    plugin: dump
+    organize:
+      urlscan-wrapper: bin/urlscan-wrapper
```

Comment:
Is the unsupported locale issue specific to a particular distro or a snap issue? Is it something I can fix in the code? How can I replicate the locale error?

Sorry, I don't know much about snaps and haven't worked with them at all before!

Reply:
To reproduce the locale error you have to build the snap using this [snapcraft.yaml](https://pastebin.com/9EbfJi2v) file.
```sh
$ cd urlscan
$ snapcraft cleanbuild
$ sudo snap install urlscan_git_amd64.snap
$ echo 'http://foo.bar' | urlscan
Traceback (most recent call last):
  File "/snap/urlscan/x1/bin/urlscan", line 187, in <module>
    main()
  File "/snap/urlscan/x1/bin/urlscan", line 172, in main
    msg = process_input(args.message)
  File "/snap/urlscan/x1/bin/urlscan", line 89, in process_input
    locale.setlocale(locale.LC_ALL, '')
  File "/snap/urlscan/x1/usr/lib/python3.5/locale.py", line 594, in setlocale
    return _setlocale(category, locale)
locale.Error: unsupported locale setting
```
I found only two things on this subject [bug #1576411](https://bugs.launchpad.net/ubuntu/+source/snapd/+bug/1576411) and [snapcraft forum](https://forum.snapcraft.io/t/ubuntu-core-16-on-raspberry2-howto-python3-programs/2830).